### PR TITLE
fix: port long running dns check windows in e2e tests

### PR DESCRIPTION
**Pull request by cnadolny**
**Reason for Change**:
adds tests for e2e scenarios in windows containers. From issue #3632, 
* Port linux tests that make sense to a windows 
* Add [after awhile DNS check tests](https://github.com/Azure/acs-engine/blob/8824c1c9f8d03f86acbe04b4eb5bef46f60b0a92/test/e2e/kubernetes/kubernetes_test.go#L850) for dns lookups for windows containers.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [ ] includes documentation
- [ ] adds unit tests
- [x] tested upgrade from previous version

**Notes**:
Reopening this issue from acs-engine:
https://github.com/Azure/acs-engine/pull/3765

### CODE REVIEW

**Comment by tariq1890** [comment]
Thanks for the PR! The Windows e2e failures don't really relate to the changes. @PatrickLang Is this the same issue we were facing before with the windows e2e tests?

File: examples/e2e-tests/kubernetes/windows/hybrid/definition.json [hunk 1]
```diff
@@ -21,7 +21,18 @@
         "count": 2,
         "vmSize": "Standard_D2_v2",
         "availabilityProfile": "AvailabilitySet",
-        "osType": "Windows"
+        "osType": "Windows",
+        "extensions": [
```
Comment (jackfrancis) [inline examples/e2e-tests/kubernetes/windows/hybrid/definition.json]: Is this change related to the new test coverage?

Comment (PatrickLang) [reply examples/e2e-tests/kubernetes/windows/hybrid/definition.json]: this file already has SSH enabled, so is there a need for WinRM too?
